src/lib/terkin/driver/hx711_sensor.py

```diff
@@ -110,22 +118,52 @@ def read(self):
         if self.loadcell is None:
             return self.SENSOR_NOT_INITIALIZED
 
+        address = self.address
+
         #log.info('Acquire reading from HX711')
-        reading = self.loadcell.get_reading()
+        readingA = self.loadcell.get_reading()
+
+        # with dualchannel 3 data sets are generated instead of one
+        # 1 - cumulated weight, no parameter, 2 - channel A reading with parameters, 3 - same for channel B
+        if self.parameter['dualchannel']:   # read channel B
+            self.loadcell.set_gain(32)      # switch to channel B
+            self.loadcell.set_scale(self.parameter['scaleB'])
+            self.loadcell.set_offset(self.parameter['offsetB'])
+            readingB = self.loadcell.get_reading()
+            self.loadcell.set_gain(self.parameter.get('gain', 128))      # switch everything back
+            self.loadcell.set_scale(self.parameter['scale'])
+            self.loadcell.set_offset(self.parameter['offset'])
+
+            # Propagate main kg value.
+            key = 'weight.{}'.format(address)
+            totalweight = readingA.kg + readingB.kg
+            effective_data = {
+                key: totalweight
+            }
```

Comment:
I haven't expected this. With "dual-channel mode", I believed values from both kinds of readings would propagate individually like outlined below.

Reply:
There is (and has been) 'scale.xxx' data which has everything and there is 'weight' which holds the result in kgs. I kept that but now the total weight is the sum of both readings. If you are only interested in the weight you can have it and it is backwards compatible.

Reply:
So I believe it will be okay then. Thanks for the explanation.